File: webapp/lib/models/event-context.ts
/**
 * Event context model
 */

export interface EventContext {
  eventName?: string;
  date?: Date;
  djPerformer?: string;
  specialNotes?: string;
}

export function hasEventInfo(context?: EventContext): boolean {
  if (!context) return false;
  return !!(
    context.eventName ||
    context.date ||
    context.djPerformer ||
    context.specialNotes
  );
}

export function formatDate(
  date: Date,
  formatStr: string = "EEEE, MMMM d"
): string {
  // Simple date formatter - for production, use date-fns or similar
  const days = [
    "Sunday",
    "Monday",
    "Tuesday",
    "Wednesday",
    "Thursday",
    "Friday",
    "Saturday",
  ];
  const months = [
    "January",
    "February",
    "March",
    "April",
    "May",
    "June",
    "July",
    "August",
    "September",
    "October",
    "November",
    "December",
  ];

  const dayName = days[date.getDay()];
  const monthName = months[date.getMonth()];
  const dayNum = date.getDate();

  return `${dayName}, ${monthName} ${dayNum}`;
}

File: nightlife_engine/models/event_context.py
"""Event context model for nightlife events."""

from dataclasses import dataclass
from typing import Optional
from datetime import datetime


@dataclass
class EventContext:
    """Context information for a specific event."""

    event_name: Optional[str] = None
    date: Optional[datetime] = None
    dj_performer: Optional[str] = None
    special_notes: Optional[str] = None

    def has_event_info(self) -> bool:
        """Check if any event information is provided."""
        return any([
            self.event_name,
            self.date,
            self.dj_performer,
            self.special_notes
        ])

    def format_date(self, format_str: str = "%A, %B %d") -> Optional[str]:
        """Format the event date."""
        if self.date:
            return self.date.strftime(format_str)
        return None
